### ISSUE TITLE
`:unquote-not-syntax-quoted` triggers false positives in `project.clj` when using defproject with shared version vars

### ISSUE THREAD

**Issue by evg-tso** [clj-kondo/clj-kondo#2695]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2025.12.23

**macro usage**

Yes. This seems related to `leiningen`’s `defproject` macro and unquote usage in `project.clj`.

**platform**

Both native version and on the JVM.

**editor**

Editor: IntelliJ
Plugin: Cursive

But also plain cli.

**problem**

After upgrading to a `clj-kondo` version that includes the new `:unquote-not-syntax-quoted` linter, I’m getting warnings in `project.clj` for a common Leiningen pattern: defining a shared version var and reusing it in dependencies via `~`.  

Example pattern:
```clojure
(def proto-version "1.2.3")

(defproject my-project "0.1.0"
  :dependencies [[some/lib ~proto-version]
                 [other/lib ~proto-version]])
```

**repro**

Minimal repo repro:
https://github.com/evg-tso/cursive-shrink-issue/blob/master/project.clj

**expected behavior**

- `:unquote-not-syntax-quoted` should not warn for valid `~` usage inside `defproject` / `project.clj`
